### ISSUE TITLE
tensorboard image support during training

### ISSUE THREAD

**Issue by pablovela5620** [open-mmlab/mmpose#225]
It would be great to be able to visualize network predictions in the evaluation portion of the training loop. Detectors already have a `show_results` method and tensorboard seems to already be support for scalar/tensor values output by `forward_test`. I'm not sure if this is already support or not, but I haven't managed to find a way to do so

**Comment by innerlee** [comment]
> ...to visualize network predictions...

How many images to visualize, and how to sample them?

**Comment by pablovela5620** [comment]
Good questions! Here are my proposals

**Number of images**
Visualization here in my opinion is mostly as a sanity check to ensure the network is actually learning what's expected. This is especially useful when overfitting on a small subset of the dataset.

 In my opinion, I think showing 3 grids (using something like torchvision.utils.make_grid)) consisting of 4/5 images would be ideal (based on sampling strategy). Any more and I think it would just overwhelm/not really add use.  

 **Sampling Strategy** 
I think the following three things should be sampled (each a grid of 4/5 images)

1. Random sampling (with set seed for consistency across experiments) just as a general view into the dataset
2. Loss Sampling: Look at samples with the highest loss values to see where the network is having trouble
3. Metric Sampling: Look at samples with the worst metrics (EPE/PCK/ect) 

Thank you again for taking this into consideration and please let me know what you think. This is just what came to mind so any feedback would be great

**Comment by innerlee** [comment]
We had plans to upgrade the visualization support in mmcv. This is a good use case, and is helpful for the new design.

The strategy described above have similarity with GANs. But in GANs, they sometimes fix a set of random numbers, to see how the quality evolves. So fixed sampling is also a good option.

**Comment by rtgunti** [comment]
Hi @innerlee 
Could you please let me know if there has been any progress on this?
